src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -435,23 +444,60 @@ def end_string(self):
 class OMPParallelDirective(OMPRegionDirective):
     ''' Class representing an OpenMP Parallel directive. '''
 
+    _children_valid_format = ("Schedule, OMPDefaultClause, [OMPPrivateClause], "
+                              "[OMPReductionClause]*")
+
+    def __init__(self, children=None, parent=None):
+       super(OMPParallelDirective, self).__init__(children=children, parent=parent)
+       self.addchild(OMPDefaultClause(clause_type=
```

Comment:
This is to mirror the current behaviour with clause generation. At the moment the `OMPParallelDirective` will always generate `!$omp parallel default(shared)`, so I add the child in the constructor to replicate that behaviour.

Reply:
The create method works. The one slight bit of weirdness is in `OMPParallelTrans` there is now:

```
         # Set the type of directive that the base class will use
         self._pdirective = OMPParallelDirective.create  
```

since we need to use the `create` method instead of the init function, but as long as you're ok with that it should work.

Reply:
Do I need to add any docs to say to use the `create` method instead of the constructor?

Reply:
I see there are some inits and some create in use now
```
src/psyclone/transformations.py
1473:        self._pdirective = None
1606:        self._pdirective = self._directive
1738:        self._pdirective = OMPMasterDirective
1793:        self._pdirective = OMPParallelDirective.create
1866:        self._pdirective = ACCParallelDirective
```

but this ended up looking as an OOP factory pattern I think. Maybe renaming the attribute to `self._directive_factory` will make the intent more clear and already imply that any method that returns a new directive can be provided. What do you think?

Also you will need to tweak the comment preceding this assignment.

Reply:
I think thats a better naming scheme. I've made this change and have updated the comment above I think.